### PR TITLE
Rename PolyMC to Prism

### PR DESCRIPTION
**Pull request by MenacingPerson**
This was useful btw:
```
$ rg --regexp '[Pp]oly[Mm][Cc]+|[Pp]rims+' --ignore-case | grep -vE '//|\*|Prism|^COPYING.md'
$ find . -type f -exec sed -i 's|PolyMC - Minecraft Launcher|Prism Launcher - Minecraft Launcher|g' {} +
$ rg --regexp '[Pp]oly[Mm][Cc]+|[Pp]rims+' --ignore-case | grep -vE '^COPYING.md|PolyMC - Minecraft Launcher|Prism'
```

### CODE REVIEW

**Comment by TheKodeToad** [comment]
I noticed it's also replaced in the LICENSE files. I wonder if this is OK or all contributors would need to give permission.

File: launcher/Application.cpp [hunk 1]
```diff
@@ -301,16 +301,16 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
         dataPath = foo.absolutePath();
         adjustedBy = "Persistent data path";
 
-        QDir polymcData(FS::PathCombine(QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation), "PolyMC"));
-        if (polymcData.exists()) {
-            dataPath = polymcData.absolutePath();
-            adjustedBy = "PolyMC data path";
+        QDir prismlauncherData(FS::PathCombine(QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation), "PrismLauncher"));
+        if (prismlauncherData.exists()) {
```
Comment (DioEgizio) [inline launcher/Application.cpp]: that's intended to be polymc

**Comment by DioEgizio** [comment]
> I noticed it's also replaced in the LICENSE files. I wonder if this is OK or all contributors would need to give permission.

yeah better to keep polymc there for now

File: launcher/Application.cpp (hunk 1 above)
Comment (MenacingPerson) [reply launcher/Application.cpp]: Okay got it

Comment (MenacingPerson) [reply launcher/Application.cpp]: I think it's done

**Comment by MenacingPerson** [comment]
> >  I noticed it's also replaced in the LICENSE files. I wonder if this is OK or all contributors would need to give permission.
> yeah better to keep polymc there for now

Technically, all PolyMC contributors are PrismLauncher contributors. I can revert easily though.

File: launcher/updater/GoUpdate.cpp [hunk 2]
```diff
@@ -104,7 +104,7 @@ bool processFileLists
         }
     }
 
-    // Next, check each file in PolyMC's folder and see if we need to update them.
+    // Next, check each file in PrismLauncher's folder and see if we need to update them.
```
Comment (Scrumplex) [inline launcher/updater/GoUpdate.cpp]:
```suggestion
    // Next, check each file in Prism Launcher's folder and see if we need to update them.
```

File: libraries/README.md [hunk 3]
```diff
@@ -173,7 +173,7 @@ Available either under LGPL version 2.1 or later.
 
 ## systeminfo
 
-A PolyMC-specific library for probing system information.
+A PrismLauncher-specific library for probing system information.
```
Comment (Scrumplex) [inline libraries/README.md]:
```suggestion
A Prism Launcher-specific library for probing system information.
```

File: launcher/java/JavaUtils.cpp [hunk 4]
```diff
@@ -441,7 +441,7 @@ QList<QString> JavaUtils::FindJavaPaths()
     scanJavaDir("/usr/lib/jvm");
     scanJavaDir("/usr/lib64/jvm");
     scanJavaDir("/usr/lib32/jvm");
-    // javas stored in PolyMC's folder
+    // javas stored in PrismLauncher's folder
```
Comment (Scrumplex) [inline launcher/java/JavaUtils.cpp]:
```suggestion
    // javas stored in Prism Launcher's folder
```

File: launcher/minecraft/Library.h [hunk 5]
```diff
@@ -215,23 +215,23 @@ class Library
     /// DEPRECATED URL prefix of the maven repo where the file can be downloaded
     QString m_repositoryURL;
 
-    /// DEPRECATED: PolyMC-specific absolute URL. takes precedence over the implicit maven repo URL, if defined
+    /// DEPRECATED: PrismLauncher-specific absolute URL. takes precedence over the implicit maven repo URL, if defined
```
Comment (Scrumplex) [inline launcher/minecraft/Library.h]:
```suggestion
    /// DEPRECATED: Prism Launcher-specific absolute URL. takes precedence over the implicit maven repo URL, if defined
```

File: launcher/minecraft/Library.h [hunk 6]
```diff
@@ -215,23 +215,23 @@ class Library
...
+    /// Prism Launcher extension - filename override
     QString m_filename;
 
-    /// DEPRECATED PolyMC extension - display name
+    /// DEPRECATED Prism Launcher extension - display name
     QString m_displayname;
 
     /**
-     * PolyMC-specific type hint - modifies how the library is treated
+     * PrismLauncher-specific type hint - modifies how the library is treated
```
Comment (Scrumplex) [inline launcher/minecraft/Library.h]:
```suggestion
     * Prism Launcher-specific type hint - modifies how the library is treated
```

File: launcher/minecraft/Library.h [hunk 7]
```diff
@@ -215,23 +215,23 @@ class Library
...
      */
     QString m_hint;
 
     /**
-     * storage - by default the local libraries folder in polymc, but could be elsewhere
-     * PolyMC specific, because of FTB.
+     * storage - by default the local libraries folder in PrismLauncher, but could be elsewhere
```
Comment (Scrumplex) [inline launcher/minecraft/Library.h]:
```suggestion
     * storage - by default the local libraries folder in Prism Launcher, but could be elsewhere
```

**Review by Scrumplex — CHANGES_REQUESTED** [review]

**Comment by Scrumplex** [comment]
Ideally, commit my suggestion in a batch

**Review by DioEgizio — COMMENTED** [review]
about renaming the images, you probably shouldn't do it since https://github.com/PrismLauncher/PrismLauncher/pull/92 already deals with that stuff

**Comment by DioEgizio** [comment]
please [sign off your commits](https://github.com/PrismLauncher/PrismLauncher/pull/54/checks)

**Comment by MenacingPerson** [comment]
I am quite new to this, sorry

**Review by Scrumplex — APPROVED** [review]
Thanks

**Review by DioEgizio — APPROVED** [review]
lgtm

**Pull request merged**